519 enhancement stub upc products

wilmelz opened the pull request:
Added ability to run script as a management command.  Added YAML type output.  Added writing of PKs to output.

debboutr commented:
this runs using the command:
* `python manage.py product_dup_check`
* need to have the package `fuzzywuzzy` installed so we should add to README
I got this warning when running, so we might add this pkg as well
```
/home/debboutr/anaconda3/envs/hem/lib/python3.7/site-packages/fuzzywuzzy/fuzz.py:11: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning
  warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')
```

wilmelz commented:
Added new package called ruamel.yaml.  This package can be installed by entering "pip install ruamel.yaml" from the terminal window. It is used for the better handling of YAML output.  Disabled warning regarding the installation of the Levenshtein package.  Added printing of output location and names of files.  Simplified the coding in the script.

debboutr commented:
do we want to return the product in the event that there are no matching UPC's?  how are these files going to be used?
```
- aquasorb; pk = 1564:
  - UPC Product:
    - No matches found.
- lithiun manganese dioxide batteries; pk = 1563:
  - UPC Product:
    - No matches found.
- ? "purex liquid laundry detergent \u2013 free & clear, he\u2013free & clear, original\
    \ fresh, mountain breeze, after the rain, he \u2013 after the rain, lavender (sunshine\
    \ clean canada), lemon fresh, baby (purex baby soft); pk = 1562"
  : - UPC Product:
      - No matches found.
```

wilmelz commented:
Added the fuzzywuzzy and ruamel.yaml packages to the requirements.txt file.

mwfrost commented on factotum/management/commands/product_dup_check.py:
```diff
@@ -0,0 +1,133 @@
+        pathname = os.getcwd()
+
+        # Open file that will contain the top match for each stub product
+        file_top_yaml = open(pathname + "\\top_match.yaml", 'w')
```
Hard-coding the backslashes causes problems for Linux/Mac users. There's a python package we found for building filesystem directories in a platform-independent way. I'll look up an example of where it's used.

mwfrost replied:
Found it: https://docs.python.org/3/library/pathlib.html

The `PurePath()` method builds platform-appropriate paths out of strings. I have not used it, but this would be the appropriate place.

Daveahern commented on dashboard/tests/functional/test_product_dup_check.py:
```diff
@@ -0,0 +1,59 @@
+        # Build list of UPC product names
+        list_upc_names = []
+        list_upc_pks = []
+        for norm_product in norm_products:
```
@wilmelz Take a look a this. The test is failing here because `norm_products` doesn't exist yet, it's instantiated at line 33.

wilmelz replied:
@Daveahern: Fixed the bug in test_product_dup_check.py and pushed the code changes into GitHub.
